apps/correlations/kinds/correlations.cue

```diff
@@ -0,0 +1,35 @@
+package kinds
+
+correlationsv0alpha1: {
+	kind:       "Correlation"  // note: must be uppercase
+	pluralName: "Correlations"
+	schema: {
+		spec: {
+			source_uid:  string
+			target_uid?:  string
+			label:       string
+			description?: string
+			config:      ConfigSpec
+			provisioned: bool
+			type:        CorrelationType
+		}
+	}
+}
+
+ConfigSpec: {
+	field: string
+	type?: string
```

Comment:
is this really:
```
{
  spec: {
     type: "query" | "external"
     config: {
       type: ???
     }
  }
}
```
what is the second type?

Do query and external share all the same properties in config?  would it make sense to have:

```
spec: {
     type: "query"
     query: {
       ...
     }
  }
````
```
spec: {
     type: "external"
     external: {
       ...
     }
  }
````